src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/VideoLinkBookingIntegrationTest.kt

```diff
@@ -64,20 +66,29 @@ import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.CourtBookingR
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.NewCourtBookingOwnerEmail
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.NewCourtBookingPrisonCourtEmail
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.NewCourtBookingPrisonNoCourtEmail
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.AppointmentCreatedEvent
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.AppointmentInformation
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.DomainEvent
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.ManageExternalAppointmentsService
 import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.OutboundEventsPublisher
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.VideoBookingCancelledEvent
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.VideoBookingCreatedEvent
+import uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.service.events.VideoBookingInformation
 import java.time.LocalDate
 import java.time.LocalTime
 import java.util.*
 
 @ContextConfiguration(classes = [TestEmailConfiguration::class])
 class VideoLinkBookingIntegrationTest : IntegrationTestBase() {
 
-  @MockBean
+  @SpyBean
   private lateinit var outboundEventsPublisher: OutboundEventsPublisher
 
   private val domainEventCaptor = argumentCaptor<DomainEvent<*>>()
 
+  @MockBean
+  private lateinit var manageExternalAppointmentsService: ManageExternalAppointmentsService
```

Comment:
This is being mocked because now we are getting events processed it is also trying to call the external services.  It is not being tested at the moment though here.

We could assert the various methods, create, cancel etc are called.

Reply:
Alternative is a mock service for each external service, but probably enough to know that the external service methods are being triggered by the inbound events.

Reply:
Yeah I know.  It's a bit of a balancing act as in how far do you take it?